ThirdParty/source/CMakeLists.txt

```diff
@@ -0,0 +1,9 @@
+set(ThirdPartySources pugixml.cpp)
+
+add_library(ThirdPartyObjects OBJECT ${ThirdPartySources})
+add_library(ThirdPartyStatic STATIC $<TARGET_OBJECTS:ThirdPartyObjects>)
```

Comment:
I'm not sure these names make sense. I doubt you will often want to link against all the third party things at once. One library for each makes more sense to me.